framework/src/main/java/org/checkerframework/framework/type/QualifierHierarchy.java

```diff
@@ -149,6 +149,33 @@ default boolean isSubtype(
         return true;
     }
 
+    /**
+     * Tests whether all qualifiers in {@code qualifiers1} are comparable with the qualifier in the
+     * same hierarchy in {@code qualifiers2}.
+     *
+     * @param qualifiers1 set of qualifiers; exactly one per hierarchy
+     * @param qualifiers2 set of qualifiers; exactly one per hierarchy
+     * @return true iff all qualifiers in {@code qualifiers1} are comparable with qualifiers in
+     *     {@code qualifiers2}
+     */
+    default boolean isComparable(
+            Collection<? extends AnnotationMirror> qualifiers1,
+            Collection<? extends AnnotationMirror> qualifiers2) {
+        assertSameSize(qualifiers1, qualifiers2);
+        for (AnnotationMirror subQual : qualifiers1) {
+            AnnotationMirror superQual = findAnnotationInSameHierarchy(qualifiers2, subQual);
+            if (superQual == null) {
+                throw new BugInCF(
+                        "QualifierHierarchy: missing annotation in hierarchy %s. found: %s",
+                        subQual, StringsPlume.join(",", qualifiers2));
+            }
+            if (!isSubtype(subQual, superQual) && !isSubtype(superQual, subQual)) {
```

Comment:
Do you mean subQual and superQual are the same?

Reply:
> Do you mean subQual and superQual are the same?

if `subQual` and `superQual` are not each other's subtype, this means two qualifiers are incomparable.

Reply:
I see. But could we just use something like `areSame(subQual, superQual)` instead of isSubtype?

Reply:
Understood.